### PR TITLE
Fix(PresenterOverlay): Add grabbing cursor

### LINKED ISSUE #11869: Use cursor:grab for the floating videostream on top of screenshares
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**

As a Nextcloud Talk user I might not discover that avatars are now draggable when someone shares their screen.

**Describe the solution you'd like**

Change the cursor to *grab* to make the feature more discoverable.

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #11869

* Moved into a new component
* Added cursor style

### 🖼️ Screenshots / Screencasts

🏡 After

https://github.com/nextcloud/spreed/assets/84044328/36696d6e-4a49-4aff-a13e-e46e0b58d01a

### 🏁 Checklist

- [ ] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/CallView/CallView.vue [hunk 1]
```diff
@@ -75,36 +75,14 @@
...
-							:title="t('spreed', 'Show presenter')"
-							class="presenter-overlay--collapse"
-							type="tertiary-no-background"
-							@click="toggleShowPresenterOverlay">
-							<template #icon>
-								<AccountBox fill-color="#ffffff" :size="20" />
-							</template>
-						</NcButton>
+						<PresenterOverlay v-if="shouldShowPresenterOverlay || isPresenterCollapsed"
+							:class="{'presenter-overlay--collapsed': isPresenterCollapsed }"
```
Comment (Antreesy) [inline src/components/CallView/CallView.vue:79]: This class is a styling for NcButton, and you know collapsed state => move to component

I see there's a hover styles on it, but I'd move them inside too with :deep selector

File: src/components/CallView/CallView.vue [hunk 2]
```diff
@@ -75,36 +75,14 @@
...
 						<!-- presenter overlay -->
-						<VueDraggableResizable v-if="shouldShowPresenterOverlay"
-							:key="presenterOverlaySize"
-							parent
-							:resizable="false"
-							:h="presenterOverlaySize"
-							:w="presenterOverlaySize"
-							:x="10"
-							:y="10">
-							<VideoVue class="presenter-overlay__video"
```
Comment (Antreesy) [inline src/components/CallView/CallView.vue:86]: This class is not moved as well

File: src/components/CallView/CallView.vue [hunk 3]
```diff
@@ -75,36 +75,14 @@
...
-							:aria-label="t('spreed', 'Show presenter')"
-							:title="t('spreed', 'Show presenter')"
-							class="presenter-overlay--collapse"
-							type="tertiary-no-background"
-							@click="toggleShowPresenterOverlay">
-							<template #icon>
-								<AccountBox fill-color="#ffffff" :size="20" />
-							</template>
-						</NcButton>
+						<PresenterOverlay v-if="shouldShowPresenterOverlay || isPresenterCollapsed"
```
Comment (Antreesy) [inline src/components/CallView/CallView.vue:78]: You can rework computed props:
`(show && isModelWithVideo) || (!show && videoAvailable)` => `isModelWithVideo || videoAvailable`

File: src/components/CallView/CallView.vue [hunk 4]
```diff
@@ -75,36 +75,14 @@
...
+							:class="{'presenter-overlay--collapsed': isPresenterCollapsed }"
+							:token="token"
+							:model="shownRemoteScreenCallParticipantModel"
+							:shared-data="sharedDatas[shownRemoteScreenPeerId]"
+							:is-collapsed="isPresenterCollapsed"
```
Comment (Antreesy) [inline src/components/CallView/CallView.vue:83]: And use data directly
```suggestion
							:is-collapsed="!showPresenterOverlay"
```

**Review by Antreesy — COMMENTED** [review]
Sorry, didn't test it, but by screencast lookg and works good)
Couple of comments:

File: src/components/CallView/CallView.vue (hunk 2 above)
Comment (DorraJaouad) [reply src/components/CallView/CallView.vue:86]: What do you mean? it is in PresenterOverlay

Comment (Antreesy) [reply src/components/CallView/CallView.vue:86]: Maybe I miss it then. Now it's all good

File: src/components/CallView/shared/PresenterOverlay.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+		&:hover,
+		&:focus {
+			opacity: 1;
+			background-color: rgba(0, 0, 0, 0.2) !important;
+		}
+	}
+}
+
+:deep(div) {
+    cursor: inherit;
```
Comment (Antreesy) [inline src/components/CallView/shared/PresenterOverlay.vue:161]: please lint new file, and add comment on that :deep selector (why it is needed)

File: src/components/CallView/shared/VideoVue.vue [hunk 6]
```diff
@@ -727,6 +727,7 @@ export default {
 	content: '';
 	box-shadow: inset 0 0 0 3px white;
 	cursor: pointer;
+	z-index: 1;
```
Comment (Antreesy) [inline src/components/CallView/shared/VideoVue.vue:730]: Not sure, how these changes affect on component, so feel free to comment it)

**Review by Antreesy — APPROVED** [review]
Tested now

**Pull request merged**